yandex/resource_yandex_kubernetes_cluster.go

```diff
@@ -361,7 +366,19 @@ func resourceYandexKubernetesClusterRead(d *schema.ResourceData, meta interface{
 		return handleNotFoundError(err, d, fmt.Sprintf("Kubernetes cluster with ID %q", clusterID))
 	}
 
-	return flattenKubernetesClusterAttributes(cluster, d, true)
+	if err := flattenKubernetesClusterAttributes(cluster, d, true); err != nil {
+		return err
+	}
+
+	response, err := config.sdk.CreateIAMToken(ctx)
+	if err != nil {
+		return err
+	}
+
+	token := response.GetIamToken()
```

Comment:
IAM token lifetime is very short (like 1 hour). If you plan to put this value somewhere in config, and keep it there, it will not do. 
Can you describe what are you trying to achieve with these changes?

Reply:
I'd like to use
```
data "yandex_kubernetes_cluster" "default" {
  cluster_id = data.terraform_remote_state.devops_base.outputs["kubernetes_cluster_id"]
}

provider "kubernetes" {
  load_config_file = false

  host                   = data.yandex_kubernetes_cluster.default.master.0.external_v4_endpoint
  cluster_ca_certificate = data.yandex_kubernetes_cluster.default.master.0.cluster_ca_certificate
  token                  = data.yandex_kubernetes_cluster.default.token
}
```

instead of

```
provider "kubernetes" {
  load_config_file = false

  exec {
    command = "yc"
    .  .  .
  }
}
```

Reply:
Using `yc` may not be obvious when there are multiple clouds in the profiles.

In addition, I find it strange to exec command-line client utility from the provider, while the provider is a self-contained Cloud API client.

Reply:
Will more acceptable to retrieve token by data source like [google_client_config](https://www.terraform.io/docs/providers/google/d/datasource_client_config.html) instead?

Reply:
yes, that looks a lot better and also reusable for other resources

Reply:
With new data source merged, I guess we can discard this PR now.